Sofa/framework/Geometry/src/sofa/geometry/Edge.h

```diff
@@ -292,8 +292,18 @@ struct Edge
                 return false;
             }
 
-            const T alpha = alphaNom / alphaDenom;
-            const T beta = (dCACD + alpha * dABCD) / dCDCD;
+            T alpha = alphaNom / alphaDenom;
+            T beta = (dCACD + alpha * dABCD) / dCDCD;
+
+            if (abs(alpha) < EQUALITY_THRESHOLD)
+                alpha = 0;
+            else if (alpha > (1 - EQUALITY_THRESHOLD) && alpha < (1+ EQUALITY_THRESHOLD))
```

Comment:
```suggestion
            else if (abs(1-alpha) < EQUALITY_THRESHOLD)
```

and the same for beta